core/trino-main/src/main/java/io/trino/sql/gen/RowConstructorCodeGenerator.java

```diff
@@ -88,4 +97,38 @@ public BytecodeNode generateExpression(BytecodeGeneratorContext context)
         block.append(context.wasNull().set(constantFalse()));
         return block;
     }
+
+    private BytecodeNode generateExpressionForLargeRows(BytecodeGeneratorContext context)
+    {
+        BytecodeBlock block = new BytecodeBlock().setDescription("Constructor for " + rowType);
+        CallSiteBinder binder = context.getCallSiteBinder();
+        Scope scope = context.getScope();
+        List<Type> types = rowType.getTypeParameters();
+
+        Variable fieldBuilders = scope.createTempVariable(BlockBuilder[].class);
+        block.append(fieldBuilders.set(invokeStatic(RowBlockBuilder.class, "createFieldBlockBuilders", BlockBuilder[].class, constantType(binder, rowType))));
+
+        // Cache local variable declarations per java type on stack for reuse
+        Map<Class<?>, Variable> javaTypeTempVariables = new HashMap<>();
+        Variable blockBuilder = scope.createTempVariable(BlockBuilder.class);
+        for (int i = 0; i < arguments.size(); ++i) {
+            Type fieldType = types.get(i);
+            Variable field = javaTypeTempVariables.computeIfAbsent(fieldType.getJavaType(), scope::createTempVariable);
+
+            block.append(blockBuilder.set(fieldBuilders.getElement(constantInt(i))));
```

Comment:
You might try removing the `blockBuilder` variable and try inlining `fieldBuilders.getElement(constantInt(i))` on both sides of the `IfStatement` to see if that results in fewer bytecode operations per field (not sure if it will or not).

Reply:
Tried this and it didnot reduce the bytecode operations per field.

Reply:
Gotcha, thanks for checking!